java/src/main/native/include/jni_utils.hpp

```diff
@@ -803,16 +823,8 @@ JNIEnv* get_jni_env(JavaVM* jvm);
 
 #define CATCH_STD(env, ret_val)                                                                    \
   catch (const std::bad_alloc &e) {                                                                \
-    JNI_THROW_NEW(env, "java/lang/OutOfMemoryError", "Could not allocate native memory", ret_val); \
-  }                                                                                                \
-  catch (const cudf::jni::jni_exception &e) {                                                      \
-    /* indicates that a java exception happened, just return so java can throw                     \
-     * it. */                                                                                      \
-    return ret_val;                                                                                \
-  }                                                                                                \
-  catch (const cudf::cuda_error &e) {                                                              \
-    JNI_THROW_NEW(env, "ai/rapids/cudf/CudaException", e.what(), ret_val);                         \
+    JNI_CHECK_THROW_NEW(env, cudf::jni::OOM_CLASS, "Could not allocate native memory", ret_val);   \
   }                                                                                                \
   catch (const std::exception &e) {                                                                \
```

Comment:
nit: Might be good to add a comment that if `jni_execption` is caught the java exception is already thrown and this will not overwrite it.